app/assets/javascripts/Components/Modals/create_modify_annotation_panel_modal.jsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import Modal from "react-modal";
+
+class CreateModifyAnnotationPanel extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      content: this.props.content,
+      category_id: this.props.category_id,
+    };
+  }
+
+  componentDidMount() {
+    Modal.setAppElement("body");
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps !== this.props) {
+      this.setState({
+        content: this.props.content,
+        category_id: this.props.category_id,
+      });
+    }
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    this.props.onSubmit({ ...this.state });
+    this.setState({ content: "", category_id: "" });
```

Comment:
This part is interesting. It seems like the parent will reset this component's `content` and `category_id` props after a submit request, making this `setState` unnecessary (I think?).